src/yukon/backend/src/api/controllers/nodes/__init__.py

```diff
@@ -1,129 +1,256 @@
-                                            990, '4.3.2.1', '0xTOOMUCHBEEF', '1.2.3.4', 'my-awesome-uid',
-                                            'I am authentic')
-
-    return jsonify(mock_responses.serialise())
-
-
-@nodes_controller.route('/<int:nodeId>/shutdown', methods=['PUT'])
-async def node_shutdown(nodeId) -> Any:
-    return "", 200
-
-
-@nodes_controller.route('/<int:nodeId>/firmwareupdate', methods=['PUT'])
-async def node_firmware_update(nodeId, param) -> Any:
-    body = await request.get_json()
-    return body['name'], 200
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019  UAVCAN Development Team  <uavcan.org>
+#               2020  dronesolutions.io. All rights reserved.
+# This software is distributed under the terms of the MIT License.
+#
+
+"""
+.. module:: controllers.nodes
+   :platform: Unix, Windows
+   :synopsis: Runs the Yukon backend
+
+.. moduleauthor:: Theodoros Ntakouris <zarkopafilis@gmail.com>
+.. moduleauthor:: Nuno Marques <nuno.marques@dronesolutions.io>
+"""
+
+import pyuavcan.transport.udp
+import pyuavcan
+import sys
+import os
+import tempfile
+import pathlib
+import asyncio
+from quart import Blueprint, jsonify, request, Response
+from typing import Any, Dict, Tuple
+
+dsdl_generated_dir = pathlib.Path(
+    tempfile.gettempdir(), 'dsdl', f'pyuavcan-v{pyuavcan.__version__}')
+dsdl_generated_dir.mkdir(parents=True, exist_ok=True)
+sys.path.insert(0, str(dsdl_generated_dir))
+
+try:
+    import pyuavcan.application             # noqa E402
+    import uavcan.node                      # noqa E402
+    import uavcan.node.Version_1_0          # noqa E402
+    import uavcan.node.Heartbeat_1_0        # noqa E402
+except (ImportError, AttributeError):
+    sys.stderr.write(
+        '\n\n\033[91mFailed to import required DSDL packages. They might not have been generated correctly\n\033[0m')
+    sys.exit(1)
+
+
+class Controller():
+    """
+        Provides utillities to bus and node monitoring and the required applicational routing
+        to the endpoints.
+    """
+
+    def __init__(self, node_list=list(), nodeid_list=list()):
+        # Blueprint backend application properties init
+        self._nodes_controller = Blueprint('nodes', __name__)
+
+        # Array of online nodes and its IDs
+        self._node_list = node_list
+        self._nodeid_list = nodeid_list
+
+        ######################
+        # ENDPOINT RESPONSES #
+        ######################
+        @self.nodes_controller.route('/', methods=['GET'])
+        async def list_of_nodes() -> Tuple[Response, None]:
+            """
+                Reponse to the home page, providing data for the existing nodes in a specific bus.
+            """
+            return jsonify([e.serialise() for e in self.node_list])
+
+        @self.nodes_controller.route('/<int:nodeId>/parameters', methods=['GET'])
+        async def node_parameter_list(nodeId) -> Tuple[Response, None]:
+            class NodeParametersResponse(object):
```

Comment:
Seems like this and other places where you are manually defining a "serialize" method you might just want to make the type extend dict. You'll then be able to use `json.dumps` to serialize the object.